enumerable.rb

```diff
@@ -0,0 +1,182 @@
+# CyclomaticComplexity Bug
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/MethodLength,Metrics/ModuleLength
+module Enumerable
+  def my_each
+    array_enum = is_a?(Range) ? to_a : self
+    result = ''
+    i = 0
+    if block_given?
+      while i < array_enum.size
+        yield(array_enum[i])
+        i += 1
+      end
+    else
+      result = array_enum.to_enum
+    end
+    !result.is_a?(Enumerator) ? array_enum : result
+  end
+
+  def my_each_with_index
+    array_enum = is_a?(Range) ? to_a : self
+    i = 0
+    result = ''
+    if block_given?
+      while i < array_enum.size
+        yield(array_enum[i], i)
+        i += 1
+      end
+    else
+      result = array_enum.to_enum :each_with_index
+    end
+    !result.is_a?(Enumerator) ? array_enum : result
+  end
+
+  def my_select
+    array_enum = is_a?(Range) ? to_a : self
+    res_array = []
+    result = ''
+    if block_given?
+      array_enum.my_each do |item|
+        res_array << item if yield(item)
+      end
+    else
+      result = array_enum.to_enum :select
+    end
+    !result.is_a?(Enumerator) ? array_enum : result
+  end
```

Comment:
my_select - returns an array containing all elements of enum for which the given block returns a true value

`[1,2,3,4,5].my_select { |num|  num.even?  }` should return `[2, 4]`
`[1,2,3,4,5].my_select { |num|  num.odd?  }` should return `[1, 3, 5]`
`[1,2,3,4,5].my_select { |num|  num > 4  }` should return `[5]`